Dockerfile.build

```diff
@@ -0,0 +1,13 @@
+FROM jenkins4eval/jenkinsfile-runner:maven
+
+# Copy the pom.xml into the image to install all dependencies so they can be cached by Codeship
+COPY pom.xml ./
+
+# Run install task so all necessary dependencies are downloaded and cached in
+# the Docker image. We're running through the whole process but disable
+# testing and make sure the command doesn't fail.
+RUN mvn install clean --fail-never -B -DfailIfNoTests=false
+
+
+# This gets the source in place for codeship to do its thing from codeship-steps.yml
+COPY . ./
```

Comment:
You do not really use it, code is passed through the workspace

Reply:
ok will get rid of it - was from an old version of mine. 

I think it is a pretty good cache - a permanent desktop like cache experience is not really ideal IMO (often you end up wanting to blow it away), or there would be a cache *server* somewhere to keep build instances ephemeral. I think it is a reasonable cloud native-like approach (and I think outside of the java ecosystem, not an uncommon way to work, eg in npm there is mostly a per project cache).

Reply:
so I think its more semantics - its fine just to say it is a cache (if it was a zip file in s3, similar challenges would result). It should be clear it isn't a cache server though